templates/hooks/bucket_sdk_file_end.go.tpl

```diff
@@ -0,0 +1,27 @@
+{{ $CRD := .CRD }}
+{{ $SDKAPI := .SDKAPI }}
+
+{{ range $specFieldName, $specField := $CRD.Config.Resources.Bucket.Fields -}}
+{{- $operationName := $specField.From.Operation }}
+{{- $path := $specField.From.Path }}
+{{- if (eq (slice $operationName 0 3) "Put") }}
+{{- $field := (index $CRD.SpecFields $specFieldName )}}
+{{- $operation := (index $SDKAPI.API.Operations $operationName) -}}
+
+{{- range $memberRefName, $memberRef := $operation.InputRef.Shape.MemberRefs -}}
+{{- if (eq $memberRef.Shape.Type "structure") }}
+
+// create{{ $memberRefName }} returns a {{ $memberRefName }} object 
```

Comment:
recommend calling this `new{{ $memberRefName }}` instead, since it's not creating (in terms of a Create API call) and you call these methods in the `newGetBucketLoggingPayload` method...